agent.go

```diff
@@ -385,6 +385,100 @@ func (c *controller) agentClose() {
 	agent.networkDB.Close()
 }
 
+type Task struct {
+	Name       string
+	EndpointID string
+	EndpointIP string
+	Info       string
+}
+
+type ServiceInfo struct {
+	VIP   string
+	Tasks []Task
+	Ports []string
+}
+type epRecord struct {
+	ep   EndpointRecord
+	info string
+}
+
+func (n *network) Services() map[string]ServiceInfo {
+	eps := make(map[string]epRecord)
+
+	if !n.isClusterEligible() {
+		return nil
+	}
+	agent := n.getController().getAgent()
+	if agent == nil {
+		return nil
+	}
+
+	// Walk through the libnetwork's "endpoint_table" and fetch the driver
+	// agnostic endpoint info
+	entries := agent.networkDB.GetTableByNetwork("endpoint_table", n.id)
+	for eid, value := range entries {
+		var epRec EndpointRecord
+		if err := proto.Unmarshal(value.([]byte), &epRec); err != nil {
+			logrus.Errorf("Unmarshal of %s failed", eid)
+			continue
+		}
+		eps[eid] = epRecord{
+			ep: epRec,
+		}
+	}
+
+	// Walk through the driver's tables, have the driver decode the entries
+	// and return the tuple {ep ID, value}. value is a string that coveys
+	// relevant info about the endpoint.
+	d, err := n.driver(true)
+	if err != nil {
+		logrus.Errorf("Can't get the driver for network, %s\n", n.ID())
+		return nil
+	}
+	for _, table := range n.driverTables {
+		entries := agent.networkDB.GetTableByNetwork(table, n.id)
+		for key, value := range entries {
+			epID, info := d.DecodeTableEntry(table, key, value.([]byte))
+			if ep, ok := eps[epID]; !ok {
+				logrus.Errorf("epRecord update attempt for driver info, but missing for %s", epID)
```

Comment:
I can remove the error message. The main purpose of the the verbose command is to expose the docker daemon's state and use it to match against the kernel state through an out of band tool, like from a separate privileged container. If the driver has any spurious info it should be reflected in the kernel state and caught by that.

In the user output lets not add something that is vague and hard to explain. Obviously it can be added later if there is a compelling reason.